strawberryfields/configuration.py

```diff
@@ -15,28 +15,136 @@
 This module contains functions used to load, store, save, and modify
 configuration options for Strawberry Fields.
 """
+import collections
 import os
 
 import toml
 from appdirs import user_config_dir
 
-from strawberryfields.logger import create_logger
 
 DEFAULT_CONFIG_SPEC = {
     "api": {
         "authentication_token": (str, ""),
         "hostname": (str, "platform.strawberryfields.ai"),
         "use_ssl": (bool, True),
         "port": (int, 443),
-    }
+    },
+    "logging": {"level": (str, "info"), "logfile": ((str, type(None)), None)},
 }
+"""dict: Nested dictionary representing the allowed configuration
+sections, options, default values, and allowed types for Strawberry
+Fields configurations. For each configuration option key, the
+corresponding value is a length-2 tuple, containing:
+
+* A type or tuple of types, representing the allowed type
+  for that configuration option.
+
+* The default value for that configuration option.
+
+.. note::
+
+    By TOML convention, keys with a default value of ``None``
+    will **not** be present in the generated/loaded configuration
+    file. This is because TOML has no concept of ``NoneType`` or ``Null``,
+    instead, the non-presence of a key indicates that the configuration
+    value is not set.
+"""
 
 
 class ConfigurationError(Exception):
     """Exception used for configuration errors"""
 
 
-def load_config(filename="config.toml", **kwargs):
+def _deep_update(source, overrides):
```

Comment:
Or comparable such that within the function it's more apparent when read
```suggestion
def _deep_update(old_dict, new_dict):
```